Lib/test/string_tests.py

```diff
@@ -1081,6 +1081,21 @@ def test_mul(self):
         # but either raises a MemoryError, or succeeds (if you have 54TiB)
         #self.checkraises(OverflowError, 10000*'abc', '__mul__', 2000000000)
 
+    def test_py3x_warnings_isinstance(self):
+        with test_support.check_py3k_warnings(("the basestring type is not supported in 3.x: "
+                                 "import a third party library like six and use"
+                                 "a compatible type like string_types", Py3xWarning)):
+            isinstance(u"fix", basestring)
+            isinstance(b"fix", basestring)
+            isinstance("fix", basestring)
+
+    def test_py3x_warnings_join(self):
+        with test_support.check_py3k_warnings(("string concatenation on mixed types is not supported in 3.x: "
+                                  "encode or decode the strings to match types", Py3xWarning)):
+            x = 'foo'
+            y = b'foo'
+            y = x + y
```

Comment:
Do we need to test all varieties of `+` here because `a + b` might not call the same code (let alone give the same result) as `b + a`?

Reply:
Good point. What if the user has over-ridden `__add__`?